tianshou/utils/logger/pandas_logger.py

```diff
@@ -0,0 +1,123 @@
+import os
+from collections import defaultdict
+from collections.abc import Callable
+from typing import Any
+
+import pandas as pd
+
+from tianshou.utils import BaseLogger, logging
+from tianshou.utils.logger.base import VALID_LOG_VALS, VALID_LOG_VALS_TYPE
+
+
+class PandasLogger(BaseLogger):
+    def __init__(
+        self,
+        log_dir: str,
+        train_interval: int = 1000,
+        test_interval: int = 1,
+        update_interval: int = 1000,
+        info_interval: int = 1,
+        exclude_arrays: bool = True,
+    ) -> None:
+        super().__init__(
+            train_interval,
+            test_interval,
+            update_interval,
+            info_interval,
+            exclude_arrays,
+        )
+        self.log_path = log_dir
+        self.csv_name = os.path.join(self.log_path, "log.csv")
+        self.pkl_name = os.path.join(self.log_path, "log.pkl")
+        self.data: dict[str, list] = defaultdict(list)
+        self.last_save_step = -1
```

Comment:
Can we make some attributes private?

Reply:
removed some of them. from the leftovers, which one should be private?

Reply:
removed the pandas logger from this PR